### PR TITLE
HZN-597 Create unit test to profile Syslogd's ConvertToEvent performance

### PR DESCRIPTION
**Pull request by MS043660**
Current progress:
-Two test cases one for single message and one for 1000 messages at a time.
-Mean time is provided in Sysout which would be removed after code review.
-Metrics are added with thread sleep inorder to visualise in console which would be removed after code review.

Further-
-Will test code using Jprofiler and would find grey areas taking more time.

### CODE REVIEW

File: features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java [hunk 1]
```diff
@@ -141,9 +144,12 @@ protected static Date parseDate(final String dateString) {
             // Date pattern has been crearted and checked inside if loop instead of 
             // parsing date inside the exception class.
             if (dateString.matches(datePattern)) {
-                final DateFormat df = new SimpleDateFormat("yyyy-MM-dd", Locale.ROOT);
-                df.setTimeZone(TimeZone.getTimeZone("UTC"));
-                return df.parse(dateString);
+            	
+              LocalDate date = LocalDate.parse(dateString);
+              DateTimeFormatter df = DateTimeFormatter.ofPattern("yyyy-MM-dd",Locale.ROOT);
+              date.format(df);
```
Comment (soleger) [inline features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java]: This is not necessary, you don't need to convert the value to a String.

File: features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java [hunk 2]
```diff
@@ -141,9 +144,12 @@ protected static Date parseDate(final String dateString) {
...
+              return Date.from(date.atStartOfDay().atZone(ZoneOffset.UTC).toInstant());
```
Comment (soleger) [inline features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java:149]: I think you can abbreviate this as:
Date.from(date.atStartOfDay(ZoneOffset.UTC).toInstant());

File: features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java (hunk 1 above)
Comment (MS043660) [inline features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java]: Is the comment regarding usage of dateformatter.I have used date formatter in Order to set it as Locale.Root to support local date,I would remove it if not required .

Comment (soleger) [inline features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogParser.java]: This call is not doing anything. It is only necessary if you want to convert the LocalDate into a String. We don't want to do this: we only need the java.util.Date format.

**Pull request closed without merging**